lib/jsdom/living/nodes/NodeList-impl.js

```diff
@@ -0,0 +1,43 @@
+"use strict";
+
+const idlUtils = require("../generated/utils.js");
+
+exports.implementation = class NodeListImpl {
+  constructor(args, privateData) {
+    if (privateData.nodes) {
+      this._list = [...privateData.nodes];
+      this._isLive = false;
+    } else {
+      this._list = [];
+      this._isLive = true;
+      this._version = -1;
+      this._element = privateData.element;
+      this._query = privateData.query;
+      this._update();
```

Comment:
Hmm, I had naively hoped we wouldn't have to do all this caching and stuff when we could just trigger on the appropriate property access. But that seems pretty dumb now; it'd slow things down way too much. Oh well. This is a nice simplification anyway.